### DISCUSSION TITLE
Start a rootless container runing systemd

### DISCUSSION THREAD

**Discussion by GongT** [containers/podman#23944] [General]
Is this possible? (note again, it's about running systemd inside container, not run podman as a service)

**Comment by rhatdan** [comment]
Yes.  Just define a quadlet for the user and activate it.

You might need to enable linger..

You could run a systemd based container with a quadlet defining a rootless container.

**Comment by sbrivio-rh** [comment]
I would suggest to start with `--privileged` and... just try doing it, debug errors one by one, or report them here. There will be some accesses to hardware devices going horribly wrong, but _maybe_ they all have workarounds.

**Comment by GongT** [comment]
My image is simple: 

```dockerfile
FROM fedora:latest
RUN dnf install systemd systemd-networkd
```

Run it will print many service failed start, `podman exec -l systemctl list-units` show there are multiple start job, everything "seems" to work. Then after random time (~1min), systemd in container core dump.

I searched around and found someone use systemd in docker without any issue, and he's using `archlinux` base image.     
I switched my base image from `fedora` to `archlink`, and it just works...

I thought it was systemd problem, but it isn't. After some test, I finally written this `entrypoint.sh`. I don't know why this works, LOL.

```bash
#!/bin/bash
shopt -s nullglob
echo "$container_uuid" >/etc/machine-id
echo "$container_uuid" >/run/machine-id

rm -rf /usr/lib/systemd/system/local-fs.target.wants /usr/lib/systemd/system/graphical.target.wants /usr/lib/systemd/system/multi-user.target.wants
rm -rf /etc/systemd/system/*.target.wants

cd /usr/lib/systemd/system/sysinit.target.wants
rm -f ./*.mount ./*.path ./systemd-firstboot.service ./systemd-machine-id-commit.service ./systemd-sysusers.service ./systemd-tpm2-* ./systemd-update-*


capsh --print
exec /usr/lib/systemd/systemd --system --log-target=console --show-status=yes --log-color=yes --crash-reboot=no
```
